src/test/scala/org/broadinstitute/dsde/vault/common/openam/OpenAMDirectivesSpec.scala

```diff
@@ -0,0 +1,236 @@
+          completeOk
+        } ~> check {
+          status should be(OK)
+          response should be(OkResponse)
+          infoMsg should endWith(" POST http://example.com/hello%n".format())
+        }
+      }
+
+      "should log the request without headers" in {
+        resetLogger()
+        Get("/hello") ~> OpenAMSession ~>
+          // NOTE: Still unsure how to simply add a header to a spray-testkit.
+          //   Based the snippet including "headerValuePF ..." from HeaderDirectivesSpec. Example:
+          //   https://github.com/spray/spray/blob/340614c9af7facf6a8b95bf4cc7733e0fca9299e/spray-routing-tests/src/test/scala/spray/routing/HeaderDirectivesSpec.scala#L81
+          `User-Agent`("custom-agent") ~> headerValuePF { case h => h } { h =>
+          logOpenAMRequest(log) {
+            completeOk
+          }
+        } ~>
+          check {
+            status should be(OK)
+            response should be(OkResponse)
+            infoMsg should endWith(" GET http://example.com/hello%n".format())
+          }
+      }
+
+      "should reject the request when missing cookie" in {
+        resetLogger()
+        Get("/hello") ~> logOpenAMRequest(log) {
+          completeOk
+        } ~> check {
+          rejection should be(MissingCookieRejection(OpenAMConfig.tokenCookie))
+        }
+      }
+
+    }
+
+    "when accessing OpenAM tokenFromCookie directive" - {
+
+      "should get a token" in {
+        Get("/hello") ~> OpenAMSession ~> tokenFromCookie { token =>
+          token shouldNot be(empty)
+          completeOk
+        } ~> check {
+          status should be(OK)
+          response should be(OkResponse)
+        }
+      }
+
+      "should reject the request when missing cookie" in {
+        Get("/hello") ~> tokenFromCookie { token =>
+          fail("should not run this line")
+          completeOk
+        } ~> check {
+          rejection should be(MissingCookieRejection(OpenAMConfig.tokenCookie))
+        }
+      }
+
+    }
+
+    "when accessing OpenAM tokenFromOptionalCookie directive" - {
+
+      "should get an optional token" in {
+        Get("/hello") ~> OpenAMSession ~> tokenFromOptionalCookie { token =>
+          token shouldNot be(empty)
+          token.get shouldNot be(empty)
+          completeOk
+        } ~> check {
+          status should be(OK)
+          response should be(OkResponse)
+        }
+      }
+
+      "should get an empty token when missing cookie" in {
+        Get("/hello") ~> tokenFromOptionalCookie { token =>
+          token should be(empty)
+          completeOk
+        } ~> check {
+          status should be(OK)
+          response should be(OkResponse)
+        }
+      }
+
+    }
+
+    "when accessing OpenAM commonNameFromCookie directive" - {
+
+      "should get the common name" in {
+        Get("/hello") ~> OpenAMSession ~> commonNameFromCookie { commonName =>
+          commonName shouldNot be(empty)
```

Comment:
we're modifying chef and jenkins anyway. Should we add the expected commonName value to the config files, just so we can test it? It's possibly overkill, but it gets us good test coverage on an important feature.